model/api/services.py

```diff
@@ -0,0 +1,179 @@
+from enum import Enum
+from functools import lru_cache
+from logging import getLogger
+
+from sqlalchemy import func
+from sqlmodel import Session, select
+
+from api.environment import config
+from api.models import (
+    Collection,
+    Resource,
+    User,
+    UserCollection,
+)
+from api.types import (
+    CollectionDto,
+    CollectionsDto,
+    Role,
+)
+
+logger = getLogger(__name__)
+
+
+@lru_cache
+def get_session() -> Session:
+    with Session(config.get_database()) as session:
+        yield session
+
+
+class NoPermissionException(Exception):
+    """Exception raised for custom error in the application."""
+
+    def __init__(self, message, error_code):
+        super().__init__(message)
+        self.error_code = error_code
+        self.message = message
+
+    def __str__(self):
+        return f"{self.message} (Error Code: {self.error_code})"
+
+
+class CollectionPermissionEnum(Enum):
+    NONE = 0
+    READ = 1
+    WRITE = 2
+    MANAGE_USERS = 3
+    MANAGE_RESOURCES = 4
+    ALL = 5
+
+
+class ResourcePermissionEnum(Enum):
+    NONE = 0
+    READ = 1
+    WRITE = 2
+    DELETE = 3
+    ALL = 4
+
+
+class UserPermissions:
+    def __init__(
+        self,
+        resource_permission: ResourcePermissionEnum,
+        collection_permission: CollectionPermissionEnum,
+    ):
+        self.resource_permission = resource_permission
+        self.collection_permission = collection_permission
+
+
+def __is_user_super_admin(user: User):
+    return user.email in config.SUPER_ADMINS
+
+
+def __get_user_system_permissions(user: User) -> UserPermissions:
+    if __is_user_super_admin(user) or user.is_admin:
+        return UserPermissions(ResourcePermissionEnum.ALL, CollectionPermissionEnum.ALL)
+    else:
+        return UserPermissions(
+            ResourcePermissionEnum.NONE, CollectionPermissionEnum.READ
+        )
+
+
+def __get_user_collection_permissions(
```

Comment:
By returning only a single value for this function we've created implicit rules like "WRITE implies READ". Can we return arrays? e.g. 

Read-only would be `[CollectionPermission.READ]`, r/w would be `[CollectionPermission.READ, CollectionPermission.WRITE]`

Reply:
And dispose of "ALL", which is similarly implicit